src/ResourceManager/KeyVault/Commands.KeyVault/Commands/NewAzureKeyVault.cs

```diff
@@ -114,12 +114,12 @@ public override void ExecuteCmdlet()
                     // This is to unblock Key Vault in Fairfax as Graph has issues in this environment.
                     WriteWarning(ex.Message);
                 }
-                if (userObjectId != Guid.Empty)
+                if (!string.IsNullOrWhiteSpace(userObjectId))
                 {
                     accessPolicy = new AccessPolicyEntry()
                     {
                         TenantId = GetTenantId(),
-                        ObjectId = userObjectId,
+                        ObjectId = Guid.Parse(userObjectId),
```

Comment:
Guid.Parse(userObjectId) => This is going to fail for ADFS where userObjectId going to be SID